ambry-clustermap/src/main/java/com.github.ambry.clustermap/AmbryDatanode.java

```diff
@@ -30,53 +26,35 @@
 /**
  * {@link DataNodeId} implementation to use within dynamic cluster managers.
  */
-class AmbryDataNode implements DataNodeId {
+abstract class AmbryDataNode implements DataNodeId {
   private final String hostName;
-  private final Port plainTextPort;
-  private final Port sslPort;
-  private final Port http2Port;
+  // exposed for subclass access
```

Comment:
Well, protected provides more access than package private, so I went with the modifier that gave the least access needed (no access needed for extenders outside of the package).

Modifier | Class | Package | Subclass | World
-- | -- | -- | -- | --
public | Y | Y | Y | Y
protected | Y | Y | Y | N
no modifier | Y | Y | N | N
private | Y | N | N | N

But I will make them protected, since it probably makes it clearer what the intent is.